test/unit/org/apache/cassandra/index/sai/cql/AllowFilteringTest.java

```diff
@@ -17,19 +17,165 @@
+                                                                                  .onMethod("startInitialBuild"))
+                                                                 .build();
+
+    @Test
+    public void testAllowFilteringDuringIndexBuild() throws Throwable
+    {
+        createTable("CREATE TABLE %s (pk text, i int, j int, k int, vec vector<float, 2>, PRIMARY KEY((pk, i), j))");
+        execute("INSERT INTO %s (pk, i, j, k, vec) VALUES ('partition1', 1, 100, 200, [0.5, 1.5])");
+        Injections.inject(blockIndexBuild);
+
+        // 1. Queries before index creation
+
+        assertInvalidThrowMessage(StatementRestrictions.REQUIRES_ALLOW_FILTERING_MESSAGE,
+                                  InvalidRequestException.class,
+                                  "SELECT * FROM %s WHERE k=0");
+        execute("SELECT * FROM %s WHERE k=0 ALLOW FILTERING");
+        assertInvalidThrowMessage(String.format(StatementRestrictions.NON_CLUSTER_ORDERING_REQUIRES_INDEX_MESSAGE, "vec"),
+                                  InvalidRequestException.class,
+                                  "SELECT * FROM %s ORDER BY vec ANN OF [1, 1] ALLOW FILTERING WITH ann_options = {}");
+
+        assertInvalidThrowMessage(String.format(StatementRestrictions.NON_CLUSTER_ORDERING_REQUIRES_INDEX_MESSAGE, "vec"),
+                                  InvalidRequestException.class,
+                                  "SELECT * FROM %s ORDER BY vec ANN OF [1, 1] ALLOW FILTERING");
+
+        String idx = createIndexAsync(String.format("CREATE CUSTOM INDEX ON %%s(k) USING '%s'", StorageAttachedIndex.class.getName()));
+        String idx2 = createIndexAsync(String.format("CREATE CUSTOM INDEX ON %%s(vec) USING '%s'", StorageAttachedIndex.class.getName()));
+
+        // 2. Queries during index build - as there is already index and AF is not an option,
+        // it just complains the index is not available from now on.
+
+        assertInvalidThrowMessage("Operation failed - received 0 responses and 1 failures: INDEX_NOT_AVAILABLE",
+                                  ReadFailureException.class,
+                                  "SELECT * FROM %s WHERE k=0");
+
+        assertInvalidThrowMessage("SAI based ORDER BY clause requires a LIMIT that is not greater than 1000. LIMIT was NO LIMIT",
+                                  InvalidRequestException.class,
+                                  "SELECT * FROM %s ORDER BY vec ANN OF [1, 1] ALLOW FILTERING WITH ann_options = {}");
+
+        assertInvalidThrowMessage("Operation failed - received 0 responses and 1 failures: INDEX_NOT_AVAILABLE",
+                                  ReadFailureException.class,
+                                  "SELECT * FROM %s ORDER BY vec ANN OF [1, 1] LIMIT 10 ALLOW FILTERING");
+
+        execute("SELECT * FROM %s WHERE k=0 ALLOW FILTERING");
+
+        // A warning was emitted because the index is not yet available
```

Comment:
There was some null pointer exception after rebase, I have to check what is the issue